pages/exercises/[lessonSlug].tsx

```diff
@@ -79,7 +51,16 @@ const Exercises: React.FC<QueryDataProps<GetAppQuery>> = ({ queryData }) => {
     { text: 'exercises', url: `/exercises/${currentLesson.slug}` }
   ]
 
-  const exercise = mockExercises[exerciseIndex]
+  const currentExercises = exercises
+    .filter(exercise => exercise?.module.lesson.slug === slug)
+    .map(exercise => ({
+      challengeName: exercise?.module.name!,
+      problem: exercise?.description!,
+      answer: exercise?.answer!,
+      explanation: exercise?.explanation!
```

Comment:
Using the `?.` and `!` operators here isn't ideal.

The reason why we need to do this is because the type definition for the exercises is `exercises: [Exercise]!` instead of `exercises: [Exercise!]!`: https://github.com/garageScript/c0d3-app/blob/master/graphql/typeDefs.ts#L9

Would it be better to change the schema to `exercises: [Exercise!]!` so that we don't have to worry about null values?

Reply:
I thought we should do that once there are some exercises. Wouldn't it throw an error when getting the exercises if there are none?

Reply:
If we changed the type definition to `exercises: [Exercise!]!` and if there are no exercises (i.e. it returns an empty array `[]`), then it shouldn't throw an error because `[]` is the type `[Exercise!]!`.

It would only throw an error if there were null values in the array (e.g. `[null, null]`).

Reply:
I added a pull request for if we want to make this change: https://github.com/garageScript/c0d3-app/pull/2336